### PR TITLE
Package sub-system unit tests

### PR DESCRIPTION
**Pull request by salman-codewar**
## Purpose changes

Issue Ref: #346 

**Package sub-system**

- [x] Added Add package unit test
- [x] Added Delete package unit test
- [x] Added Get package unit test
- [x] Added Update package unit test
- [x] Added Repo function unit test
- [x] Added validation functions unit tests

**How To?**

To run tests, navigate to `package` dir and run below commands

```go
make test
```

**Demo**
pkg_test

**Coverage**
base-rate-coverage

### CODE REVIEW

File: systems/data-plan/package/pkg/db/package_repo.go [hunk 1]
```diff
@@ -22,10 +22,16 @@ func NewPackageRepo(db sql.Db) *packageRepo {
...
 
-func (r *packageRepo) Add(_package *Package, nestedFunc ...func() error) error {
-	err := r.Db.ExecuteInTransaction(func(tx *gorm.DB) *gorm.DB {
-		return tx.Create(_package)
-	}, nestedFunc...)
+func (r *packageRepo) Add(_package *Package) error {
+	err := r.Db.GetGormDb().Transaction(func(tx *gorm.DB) error {
+		result := tx.Create(_package)
+
+		if result.Error != nil {
```
Comment (delvatt) [inline systems/data-plan/package/pkg/db/package_repo.go:29]:
```go
if result.Error != nil {
			return result.Error
		}

		return nil
```
Then we can just  `return restult.Error`

File: systems/data-plan/package/pkg/db/package_repo.go [hunk 2]
```diff
@@ -40,22 +46,21 @@ func (p *packageRepo) Get(orgId, id uint64) ([]Package, error) {
 	return packages, nil
 }
 
-func (r *packageRepo) Delete(orgId , packageId uint64, nestedFunc ...func() error) error {
-	err := r.Db.ExecuteInTransaction(func(tx *gorm.DB) *gorm.DB {
-		d := tx.Delete(&Package{}, "id = ? AND org_id = ?", packageId, orgId)
+func (r *packageRepo) Delete(orgId uint64, packageId uint64) error {
+	err := r.Db.GetGormDb().Transaction(func(tx *gorm.DB) error {
```
Comment (delvatt) [inline systems/data-plan/package/pkg/db/package_repo.go:50]: Again, we can skip the transaction and  go directly to the `Where` clause.

File: systems/data-plan/package/pkg/db/package_repo_test.go [hunk 3]
```diff
@@ -0,0 +1,219 @@
+package db
```
Comment (delvatt) [inline systems/data-plan/package/pkg/db/package_repo_test.go:1]: Same comments as for the Base rates sub system  repo tests.

File: systems/data-plan/package/pkg/server/package_test.go [hunk 4]
```diff
@@ -0,0 +1,164 @@
+package server
```
Comment (delvatt) [inline systems/data-plan/package/pkg/server/package_test.go:1]: Same comments as for the Base rates sub system repo tests.

File: systems/data-plan/package/pkg/db/package_repo.go [hunk 5]
```diff
@@ -22,10 +22,16 @@ func NewPackageRepo(db sql.Db) *packageRepo {
 	}
 }
 
-func (r *packageRepo) Add(_package *Package, nestedFunc ...func() error) error {
-	err := r.Db.ExecuteInTransaction(func(tx *gorm.DB) *gorm.DB {
-		return tx.Create(_package)
-	}, nestedFunc...)
+func (r *packageRepo) Add(_package *Package) error {
```
[the remark this answers is not in the document]
Comment (salman-codewar) [reply systems/data-plan/package/pkg/db/package_repo.go:25]: Addressed

File: systems/data-plan/package/pkg/db/package_repo.go (hunk 1 above)
Comment (salman-codewar) [reply systems/data-plan/package/pkg/db/package_repo.go:29]: Addressed

File: systems/data-plan/package/pkg/db/package_repo.go (hunk 2 above)
Comment (salman-codewar) [reply systems/data-plan/package/pkg/db/package_repo.go:50]: addressed

File: systems/data-plan/package/pkg/server/package_test.go (hunk 4 above)
Comment (salman-codewar) [reply systems/data-plan/package/pkg/server/package_test.go:1]: addressed

File: systems/data-plan/package/pkg/db/package_repo_test.go (hunk 3 above)
Comment (salman-codewar) [reply systems/data-plan/package/pkg/db/package_repo_test.go:1]: addressed

File: systems/data-plan/package/pkg/db/package_repo.go [hunk 6]
```diff
@@ -22,12 +22,13 @@ func NewPackageRepo(db sql.Db) *packageRepo {
...
+	result := r.Db.GetGormDb().Create(_package)
+	if result.Error != nil {
```
Comment (delvatt) [inline systems/data-plan/package/pkg/db/package_repo.go:27]: That if statement is not necessary.

File: systems/data-plan/package/pkg/db/package_repo.go (hunk 2 above)
Comment (delvatt) [reply systems/data-plan/package/pkg/db/package_repo.go]: We are still using a transaction here.  It is not necessary here, but  we can keep it if you like.

**Review by delvatt — APPROVED** [review]

**Comment by delvatt** [comment]
We can go now. If I  see  any other things of matter, I'll open separate issues.

**Pull request merged**